src/components/buttons/AddToPlayListBtn.js

```diff
@@ -0,0 +1,148 @@
+    const [newPlayListName, setNewPlayListName] = useState('');
+
+    const {
+        state: { playLists },
+        dispatch,
+    } = useData();
+
+    function addVideoToPlayList(playList) {
+        toast.success(`Video removed from ${playList.name}`);
+        dispatch({
+            type: 'REMOVE_VIDEO_FROM_PLAYLIST',
+            payload: { name: playList.name, video: video },
+        });
+    }
+
+    function removeVideoFromPlayList(playList) {
+        toast.success(`Video added to ${playList.name}`);
+        dispatch({
+            type: 'ADD_VIDEO_TO_PLAYLIST',
+            payload: { name: playList.name, video: video },
+        });
+    }
+
+    async function addNewPlayList() {
+        if (playLists.find((list) => list.name === newPlayListName)) {
+            toast.error('Play List Already Exists, try using another name');
+            return;
+        }
+        setNewPlayListName('');
+        dispatch({
+            type: 'CREATE_NEW_PLAYLIST',
+            payload: { name: newPlayListName, videos: [] },
+        });
+    }
+
+    function isVideoInPlayList(playList) {
+        if (
+            playList.videos.find((item) => item.videoId === video.videoId) !==
+            undefined
+        ) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    async function handleCheckBoxChange(playList) {
+        try {
+            if (isVideoInPlayList(playList)) {
+                addVideoToPlayList(playList);
+            } else {
+                removeVideoFromPlayList(playList);
+            }
+        } catch (error) {
+            console.log(error);
+        }
+    }
+
+    return (
+        <div
+            id="modal"
+            className=" flex justify-center items-center flex-col
+            fixed w-screen h-screen bg-gray-800 
+            bg-opacity-50 top-0 right-0"
+            onClick={(e) =>
+                e.target.id === 'modal' ? setShowPlayListModal(false) : null
+            }
+        >
+            <div className="bg-gray-100 rounded py-2 px-4">
+                <div className="flex justify-between items-center">
+                    <h1>Add To Playlist...</h1>
+                    <button
+                        onClick={() => setShowPlayListModal(false)}
+                        className="text-2xl"
+                    >
+                        &times;
+                    </button>
+                </div>
+
+                <div className="flex flex-col divide-y divide-gray-400">
+                    {playLists.map((playList) => (
+                        <label
+                            key={playList.id}
+                            className="flex items-center justify-between py-2 cursor-pointer"
+                        >
+                            <input
+                                type="checkbox"
+                                className="mr-6"
+                                onChange={() => handleCheckBoxChange(playList)}
```

Comment:
I have a suggestiong, handleCheckBoxChange does not convey action going to be happen on click, we can change this name to more precise one.